Resources/Private/Templates/Core11/DataSheetImport/Index.html

```diff
@@ -0,0 +1,74 @@
+<html
+    xmlns:f="http://typo3.org/ns/TYPO3/Fluid/ViewHelpers"
+    xmlns:be="http://typo3.org/ns/TYPO3/CMS/Backend/ViewHelpers"
+    xmlns:core="http://typo3.org/ns/TYPO3/CMS/Core/ViewHelpers"
+    xmlns="http://www.w3.org/1999/xhtml"
+    lang="en"
+    f:schemaLocation="https://fluidtypo3.org/schemas/fluid-master.xsd"
+    data-namespace-typo3-fluid="true"
+>
+<f:layout name="Default" />
+
+<f:section name="Content">
+    <h1><f:translate key="module_name" extensionName="xlsimport" /></h1>
+    <f:flashMessages queueIdentifier="core.template.flashMessages" />
+    <f:be.infobox title="{f:translate(key:'importinfo', extensionName: 'xlsimport')}" state="-1">
+        <f:translate key="importinfo_format" extensionName="xlsimport" />
+        <ul>
+            <li>Excel 2007 (.xlsx)</li>
+            <li>Excel5 (.xls)</li>
+            <li>Excel2003XML (.xml)</li>
+            <li>OpenDocument Calc (.ods)</li>
+            <li>Gnumeric (.gnumeric)</li>
+            <li>CSV (.csv/.txt)</li>
+        </ul>
+    </f:be.infobox>
+    <form
+        action="{f:be.uri(route: 'web_xlsimport', parameters: '{action: \'upload\', id: \'{page}\'}')}"
+        enctype="multipart/form-data"
+        method="post"
+    >
+        <div class="typo3-TCEforms">
+            <fieldset class="form-section">
+                <div class="row">
+                    <div class="form-group col col-sm-6">
+                        <label class="t3js-formengine-label" for="table">
+                            <f:translate key="tableselect" extensionName="xlsimport" />
+                        </label>
+                        <select
+                            class="form-select"
+                            name="table"
+                            id="table"
+                        >
+                            <f:for each="{allowedTables}" as="tableLabel" key="allowedTable">
+                                <option value="{allowedTable}">{tableLabel}</option>
+                            </f:for>
+                        </select>
+                    </div>
+                </div>
+                <div class="row">
+                    <div class="form-group col col-sm-12">
+                        <label class="t3js-formengine-label" for="fileUpload">
+                            <f:translate key="fileselect" extensionName="xlsimport" />
```

Comment:
I would suggest to use the full language file identifier here and for all other translation viewhelper calls

So instead of

```xhtml
<f:translate key="fileselect" extensionName="xlsimport" />
```

use

```xhtml
<f:translate key="LLL:EXT:xlsimport/Resources/Private/Language/locallang.xlf:fileselect" extension="xlsimport" />
```

This helps with directly finding the correct file the key is defined - or should be defined. Beside that, it avoids some magic and early short circuit in some core classes. Sadly, if the `extensionName` is left out, it would crash with v12+ if used in non extbase context.

Reply:
At the moment, I guess, we are fine with this annotation here :)